internal/tiltfile/k8s.go

```diff
@@ -800,35 +802,54 @@ func (f portForward) Hash() (uint32, error) {
 func intToPortForward(i starlark.Int) (portForward, error) {
 	n, ok := i.Int64()
 	if !ok {
-		return portForward{}, fmt.Errorf("portForward value %v is not representable as an int64", i)
+		return portForward{}, fmt.Errorf("portForward port value %v is not representable as an int64", i)
 	}
 	if n < 0 || n > 65535 {
-		return portForward{}, fmt.Errorf("portForward value %v is not in the range for a port [0-65535]", n)
+		return portForward{}, fmt.Errorf("portForward port value %v is not in the valid range [0-65535]", n)
 	}
 	return portForward{local: int(n)}, nil
 }
 
+const ipReStr = `^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$`
+const hostnameReStr = `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`
+
+var validHost = regexp.MustCompile(ipReStr + "|" + hostnameReStr)
+
 func stringToPortForward(s starlark.String) (portForward, error) {
-	parts := strings.SplitN(string(s), ":", 2)
-	local, err := strconv.Atoi(parts[0])
+	parts := strings.SplitN(string(s), ":", 3)
```

Comment:
where did this syntax come from?

it's probably ok, but seems different from the `kubectl port-forward` syntax (which has a separate `--address` flag). https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/portforward/portforward.go#L87

i guess the advantage of the separate address flag is that you can specify multiple

Reply:
Just as with the previous discussion, not sure what the exact origin of this syntax is, but it seems to be the most commonly used way to specify hostname for port forwarding, one example is OpenSSH.
You're right, it doesn't allow for multiple addresses, so if we need to support that, we should probably change syntax to something else. That being said, in my experience there's barely ever a need to bind to multiple addresses, you usually want it to either be accessible on a specific network, or accessible on every network (which can be done by specifying `0.0.0.0` as the address).